### PR TITLE
chore(dev-deps): update @sage/design-tokens

### PR DESCRIPTION
**Pull request by jamime**
### Proposed behaviour

Update @sage/design-tokens.

### Current behaviour

@sage/design-tokens is outdated.

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Additional context

@sage/design-tokens is a peer dependency and the exact version is specified by our consumers. There have been some changes that we would like to see the impact of.

### Testing instructions

Review chromatic.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -70,7 +70,7 @@
     "@commitlint/config-conventional": "^11.0.0",
     "@cypress/react": "^5.12.0",
     "@cypress/webpack-dev-server": "^1.8.4",
-    "@sage/design-tokens": "^2.3.0",
+    "@sage/design-tokens": "^2.23.0",
```
Comment (samtjo) [inline package.json:73]: We'll need to update the peerDependencies above with the same thing

**Review by mkrds — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (mkrds) [reply package.json]: I'm just curious, why do we have these in peer-dependencies as well?

Comment (samtjo) [reply package.json]: This package provides the design tokens, so if users don't have it installed then our components will look for the Design Token CSS variables and they won't be there. So users need to have this installed as well 👍🏻

Comment (jamime) [reply package.json]: We're compatible with any version `>2.3.0, <3.0.0` so we don't need to update the peer dependency.

**Review by mkrds — APPROVED** [review]

**Review by samtjo — APPROVED** [review]

**Pull request merged**